### DISCUSSION TITLE
Feature-Request: Allow "file" scope Indentifier to be used on all kind of Members

### DISCUSSION THREAD

**Discussion by CodingMadness** [dotnet/csharplang#6516] [General]
Pls implement the possibility for types/properties to be **ONLY accessible within a file** but not outside, so when i mark a part of a member, as an example with the new "file" access- restriction modifier, the given member can ONLY be "get" or "set" OR if the file-keyword is set on the entire member, then its only visible, by types who are in the same file! 

Consider this example:

```
namespace TestSLN;

//MyTest.cs  
public class Test
{
      file int Number {get; set;}.            //this thing will only be visible in the File, lets call it, **MyTest.cs** !
      public int ID {get; file set;}           //the ID property will only be settable inside **MyTest.cs** but not outside!
      
      file void DoWork()                       //this thing will only be visible in MyTest.cs!
      {
              Console.WriteLin("Do something cool here...!");
      }
}
public class Test2
{
      public void DoMoreWork()
      {
             Test test1 = new();
              test1.DoWork();
             Console.WriteLine("here we do more cool stuff..!");
      }
}


//Program.cs 
public class Program
{
   
    public Test _test;

    void Main() 
   {
          _test.ID = 123;   //Compiletime-Error!
          _test.DoWork(); //Compiletime-Error!
          Console.WriteLine("My ID is:  " + _test.ID);    //works!
   }
}

```

Would be really cool to have ur opinion on this dear .NET team!

Cheers!

**Comment by pinkfloydx33** [comment]
https://github.com/dotnet/csharplang/issues/6011

**Reply by CodingMadness** [reply]
Sadly tho the member thing is still not possible and does not look like they even plan to add it :( Feelsbadman

**Reply by CyrusNajmabadi** [reply]
@Shpendicus We are starting with file-scoped types.  We haven't made any determinations beyond that.

**Reply by CodingMadness** [reply]
@CyrusNajmabadi I mean its a good direction but I hope u did not throw the idea out of the window to extend that to members. Its really damn useful! Because u really do not want to make always an Assembly for, when u really have just some classes/members who shall not visible everywhere but just their files.

**Reply by CyrusNajmabadi** [reply]
As i said, we haven't made any further determinations.  :)

**Reply by RikkiGibson** [reply]
The feature was definitely designed with the thinking that it may eventually be expanded to work on more kinds of members.
